### ISSUE TITLE
How to call this hook conditionally?

### ISSUE THREAD

**Issue by goodwin64** [jacobbuck/react-beforeunload#9]
Hi! I want to add `beforeunload` popup only when the form on the page was touched, like the following:

```jsx
  React.useEffect(() => {
    if (props.anyTouched) {
      useBeforeunload(onBeforeunload);
    }
  }, [props.anyTouched]);
```

How may I handle it?

**Comment by jacobbuck** [comment]
Hooks cannot be conditionally, however you can put your conditional inside the hook. For example:

```js
  useBeforeunload(event => {
    if (props.anyTouched) {
      return onBeforeunload(event);
    }
  });
```

**Comment by ddcech** [comment]
where is the `onBeforeunload()` function coming from?

**Comment by jacobbuck** [comment]
@ddcech it will be defined somewhere before the hook call.

**Comment by rsnsmf** [comment]
@jacobbuck  can you give an example of how to define onBeforeunload() ?

**Comment by goodwin64** [comment]
@rsnsmf 
```typescript
// onBeforeunload.ts

const onBeforeunload = (e: BeforeUnloadEvent) => {
  e.preventDefault();
  e.returnValue = '';
  return true;
};

export default onBeforeunload;
```

**Comment by jacobbuck** [comment]
`onBeforeunload` is just a function defined in **goodwin64**'s example. You can also put your logic inside your callback function for `useBeforeunload` without defining another function.

Also react-beforeunload handles some browser inconsistencies so you don't need assign `event.returnValue`. As long as you either return a string (even `''`) or call `event.preventDefault()`, this library will handle the rest.

**Comment by Poky85** [comment]
Btw this approach is incorrect. Event handler should be defined conditionally. Otherwise it breaks [back/forward cache](https://web.dev/bfcache/).

See docs https://developer.chrome.com/blog/page-lifecycle-api/#the-beforeunload-event

This prevents me from using this hook at all

**Comment by jacobbuck** [comment]
> Btw this approach is incorrect. Event handler should be defined conditionally. Otherwise it breaks [back/forward cache](https://web.dev/bfcache/).

Good to know! That article was very interesting, thanks for sharing 😄 

> This prevents me from using this hook at all

You can still render the [`<Beforeunload>` component](https://github.com/jacobbuck/react-beforeunload#beforeunload-component) conditionally, which will remove the event listener when removed. Or even write your own component with the hook.

I will look into adding proper conditional listening to the hook for the next version.

**Comment by jacobbuck** [comment]
I've just published v2.6.0 which supports conditional listening in the [`useBeforeunload` hook](https://github.com/jacobbuck/react-beforeunload#usebeforeunload-hook-recommended).

It will only add the `beforeunload` event listener if the passed hander is a function.
